api/incubator/src/main/java/io/opentelemetry/api/incubator/common/ArrayBackedExtendedAttributesBuilder.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.api.incubator.common;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Predicate;
+
+class ArrayBackedExtendedAttributesBuilder implements ExtendedAttributesBuilder {
+  private final List<Object> data;
+
+  ArrayBackedExtendedAttributesBuilder() {
+    data = new ArrayList<>();
+  }
+
+  ArrayBackedExtendedAttributesBuilder(List<Object> data) {
+    this.data = data;
+  }
+
+  @Override
+  public ExtendedAttributes build() {
+    // If only one key-value pair AND the entry hasn't been set to null (by #remove(AttributeKey<T>)
+    // or #removeIf(Predicate<AttributeKey<?>>)), then we can bypass sorting and filtering
+    if (data.size() == 2 && data.get(0) != null) {
+      return new ArrayBackedExtendedAttributes(data.toArray());
+    }
+    return ArrayBackedExtendedAttributes.sortAndFilterToAttributes(data.toArray());
```

Comment:
The sorting feels like part of the building, so what about moving the method into the builder class instead?

Reply:
Yeah this whole function is essential a named constructor, since the constructor itself was private.

For what its worth, this whole class is a "copy / paste / modify" of [ArrayBackedAttributes](https://github.com/open-telemetry/opentelemetry-java/blob/main/api/all/src/main/java/io/opentelemetry/api/common/ArrayBackedAttributes.java#L62), so I didn't look too critically at the implementation.

What do you think about a followup PR where I cleanup minor pattern quirks like this in both ArrayBackedAttributes and ExtendedArrayBackedAttributes?